controllers/toolchaincluster/toolchaincluster_controller.go

```diff
@@ -66,18 +68,14 @@ func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.
 	if err := r.addToolchainClusterRoleLabelFromType(ctx, toolchainCluster); err != nil {
 		return reconcile.Result{}, err
 	}
-
 	return reconcile.Result{}, r.clusterCacheService.AddOrUpdateToolchainCluster(toolchainCluster)
 }
 
 func (r *Reconciler) addToolchainClusterRoleLabelFromType(ctx context.Context, toolchainCluster *toolchainv1alpha1.ToolchainCluster) error {
 	logger := log.FromContext(ctx)
 
-	if clusterType, found := toolchainCluster.Labels[cluster.LabelType]; !found {
-		logger.Info("cluster `type` label not found, unable to add toolchain cluster role label from type")
-		return nil
-	} else if clusterType != string(cluster.Member) {
-		logger.Info("cluster `type` is not member, skipping cluster role label setting")
+	if !r.useClusterRL {
+		logger.Info("Skiping the cluster role label setting as not running in host cluster")
 		return nil
 	}
```

Comment:
I would drop this function completely. IIRC, this is a leftover from the time when we introduced the labels and we wanted to populate all existing ToolchainCluster CRs with this label in the already running instances. 
The labels is set when the ToolchainCluster is created for all member CRs https://github.com/codeready-toolchain/toolchain-cicd/blob/34ea85fe21ae05c9bb1a0a2bbdc986b94c8d9e92/scripts/add-cluster.sh#L387-L400 plus we are moving this piece of configuration to SpaceProvisionerConfig anyway.